### DISCUSSION TITLE
Remove zlib-fallback for IronPython

### DISCUSSION THREAD

**Discussion by MartinThoma** [py-pdf/pypdf#863] [Ideas]
Within [filters.py](https://github.com/py-pdf/PyPDF2/blob/main/PyPDF2/filters.py#L73-L127) we have a fallback for IronPython in case zlib is not present. This part of the code is still from the initial commit in 2011. So it might be even older than 11 years.

However, it seems as [IronPython has zlib](https://ironpython-test.readthedocs.io/en/latest/library/zlib.html). I guess it was added in the past years. `zlib` itself has been there since Python 2.3 ([PEP 273](https://peps.python.org/pep-0273/))

Can somebody please confirm with IronPython that `import zlib` works? I would like to remove the fallback as I cannot test it and it seems unnecessary.

**Comment by MartinThoma** [comment]
Let's see if Twitter can help: https://twitter.com/_martinthoma/status/1523189977265295360

**Reply by MasterOdin** [reply]
It looks like IronPython has theoretically had zlib support since Nov 2010: https://github.com/IronLanguages/ironpython2/commit/06445670fb92a186460f14a4e9e5d590bde02c8f

I would rip it out, and at the moment, I think the 2.0 branch is going to break compat with IronPython anyway as you utilize 3.6 features not yet available there.

**Reply by slozier** [reply]
Yes, `import zlib` works with IronPython (since at least 2.7.0) so no point in keeping the fallback around.
